tools/circt-verilog/circt-verilog.cpp

```diff
@@ -367,24 +406,73 @@ static LogicalResult executeWithSources(MLIRContext *context,
   }
 
   // Print the final MLIR.
+  auto outputTimer = ts.nest("MLIR Printer");
   module->print(outputFile->os());
   outputFile->keep();
   return success();
 }
 
 static LogicalResult execute(MLIRContext *context) {
+  // Abort if there are no input files to be processed.
+  if (opts.inputFilenames.empty()) {
+    WithColor::error() << "no input files\n";
+    return failure();
+  }
```

Comment:
If we don't allow it to be empty, we should mark the LLVM option as required.
Also, it'd be nice to still support input from stdin. Does it work if we give it "-" as filename?
Maybe only fail here if `format` is also not given? Otherwise, we can process from stdin by default with the specified format?

Reply:
Yeah "-" works here. We could default to reading from stdin if format is present. :+1: